modules/service/src/main/scala/lucuma/odb/graphql/mapping/MutationMapping.scala

```diff
@@ -512,19 +508,21 @@ trait MutationMapping[F[_]] extends Predicates[F] {
   private lazy val UpdateDatasets: MutationField =
     MutationField("updateDatasets", UpdateDatasetsInput.binding(Path.from(DatasetType))) { (input, child) =>
       services.useTransactionally {
-        // Our predicate for selecting datasets to update
-        val filterPredicate = and(List(
-          Predicates.dataset.observation.program.isWritableBy(user),
-          input.WHERE.getOrElse(True)
-        ))
+        requireStaffAccess {
```

Comment:
I wonder if we'll need, at some point, _either_ staff or service.  We don't need it now, but can something like that be expressed?  I suppose that's just `Staff` since `Service` is more powerful than `Staff`?

Reply:
`Guest` < `Pi` < `Ngo` < `Staff` < `Admin` < `Service` so yeah you just ask for the bigger one. We could get rid of the transitive aspect and ask for a union of roles if we need it to be more fine-grained.